src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,70 @@
 package core.basesyntax;
 
+import static java.lang.Integer.parseInt;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    public static final String SPACE_REGEX = " ";
+    public static final String BUY = "buy";
+    public static final String SUPPLY = "supply";
+    public static final String COMA_REGEX = ",";
+    public static final String RESULT_NAME = "result";
+    public static final String EMPTY_REGEX = "";
+    public static final String FILTER_PATTERN = "\\D+";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        writeDataToFile(getReport(readDataFromFile(fromFileName)), toFileName);
+    }
+
+    private String readDataFromFile(String fromFileName) {
+        File readFile = new File(fromFileName);
+        StringBuilder builder = new StringBuilder();
+        try (BufferedReader reader = new BufferedReader(new FileReader(readFile))) {
+            while (reader.ready()) {
+                builder.append(reader.readLine()).append(SPACE_REGEX);
+            }
+        } catch (IOException exception) {
+            throw new RuntimeException("Can't read data from file " + readFile, exception);
+        }
+        return builder.toString();
+    }
+
+    private String getReport(String data) {
+        int result;
+        int totalSupply = 0;
+        int totalBuy = 0;
+        StringBuilder builder = new StringBuilder();
+        String[] arr = data.split(SPACE_REGEX);
+        for (String tips : arr) {
+            final int intValue = parseInt(tips.replaceAll(FILTER_PATTERN, EMPTY_REGEX));
+            if (tips.contains(BUY)) {
+                totalBuy += intValue;
+            } else if (tips.contains(SUPPLY)) {
+                totalSupply += intValue;
+            }
```

Comment:
Here you need to use split() method to get an array of 2 values: the first is operation type, and the second is amount.
Also, create constants for array indexes